### PR TITLE
Bug in AutoDocumentEncoder where `l2_norm` argument not being applied.

### PR DESCRIPTION
**Pull request by sieu-n**

### CODE REVIEW

**Comment by sieu-n** [comment]
```
from sklearn.preprocessing import normalize
import numpy as np
v = np.array([[0.5, 0.5]])
normalize(v), v

Out: 
(array([[0.70710678, 0.70710678]]), array([[0.5, 0.5]]))
```

`sklearn.preprocessing.normalize` is not inplace operation.

**Comment by sieu-n** [comment]
@lintool @MXueguang

**Review by MXueguang — APPROVED** [review]
